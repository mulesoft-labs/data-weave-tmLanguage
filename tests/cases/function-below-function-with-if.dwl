
if (2>3) "" else ""

fun repeat(text: String, times: Number): String = if (2>3) "" else ""

/**
* Helper function that enables `repeat` to work with a `null` value.
*/
fun repeat(text: Null, times: Any): Null = null