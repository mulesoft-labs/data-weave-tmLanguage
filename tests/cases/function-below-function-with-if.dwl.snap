>
>if (2>3) "" else ""
#^^^ source.data-weave meta.if.dw keyword.control.if.dw
#   ^ source.data-weave meta.if.dw
#    ^ source.data-weave meta.if.dw constant.numeric.dw
#     ^ source.data-weave meta.if.dw keyword.operator.comparison.dw
#      ^ source.data-weave meta.if.dw constant.numeric.dw
#       ^ source.data-weave meta.if.dw
#        ^ source.data-weave
#         ^ source.data-weave string.quoted.double.dw punctuation.definition.string.begin.dw
#          ^ source.data-weave string.quoted.double.dw punctuation.definition.string.end.dw
#           ^ source.data-weave
#            ^^^^ source.data-weave keyword.control.dw
#                ^ source.data-weave
#                 ^ source.data-weave string.quoted.double.dw punctuation.definition.string.begin.dw
#                  ^ source.data-weave string.quoted.double.dw punctuation.definition.string.end.dw
>
>fun repeat(text: String, times: Number): String = if (2>3) "" else ""
#^^^ source.data-weave meta.directive.fun.dw storage.type.dw
#   ^ source.data-weave meta.directive.fun.dw
#    ^^^^^^ source.data-weave meta.directive.fun.dw entity.name.function.dw
#          ^ source.data-weave meta.directive.fun.dw
#           ^^^^ source.data-weave meta.directive.fun.dw variable.parameter.dw
#               ^ source.data-weave meta.directive.fun.dw keyword.operator.declaration.dw
#                ^ source.data-weave meta.directive.fun.dw
#                 ^^^^^^ source.data-weave meta.directive.fun.dw support.class.dw
#                       ^^ source.data-weave meta.directive.fun.dw
#                         ^^^^^ source.data-weave meta.directive.fun.dw variable.parameter.dw
#                              ^ source.data-weave meta.directive.fun.dw keyword.operator.declaration.dw
#                               ^ source.data-weave meta.directive.fun.dw
#                                ^^^^^^ source.data-weave meta.directive.fun.dw support.class.dw
#                                      ^ source.data-weave meta.directive.fun.dw
#                                       ^ source.data-weave meta.directive.fun.dw keyword.operator.declaration.dw
#                                        ^ source.data-weave meta.directive.fun.dw
#                                         ^^^^^^ source.data-weave meta.directive.fun.dw support.class.dw
#                                               ^ source.data-weave meta.directive.fun.dw
#                                                ^ source.data-weave meta.directive.fun.dw keyword.operator.assignment.dw
#                                                 ^ source.data-weave
#                                                  ^^^ source.data-weave meta.if.dw keyword.control.if.dw
#                                                     ^ source.data-weave meta.if.dw
#                                                      ^ source.data-weave meta.if.dw constant.numeric.dw
#                                                       ^ source.data-weave meta.if.dw keyword.operator.comparison.dw
#                                                        ^ source.data-weave meta.if.dw constant.numeric.dw
#                                                         ^ source.data-weave meta.if.dw
#                                                          ^ source.data-weave
#                                                           ^ source.data-weave string.quoted.double.dw punctuation.definition.string.begin.dw
#                                                            ^ source.data-weave string.quoted.double.dw punctuation.definition.string.end.dw
#                                                             ^ source.data-weave
#                                                              ^^^^ source.data-weave keyword.control.dw
#                                                                  ^ source.data-weave
#                                                                   ^ source.data-weave string.quoted.double.dw punctuation.definition.string.begin.dw
#                                                                    ^ source.data-weave string.quoted.double.dw punctuation.definition.string.end.dw
>
>/**
#^^ source.data-weave comment.block.dw punctuation.definition.comment.dw
#  ^^ source.data-weave comment.block.dw
>* Helper function that enables `repeat` to work with a `null` value.
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.data-weave comment.block.dw
>*/
#^^ source.data-weave comment.block.dw punctuation.definition.comment.dw
>fun repeat(text: Null, times: Any): Null = null
#^^^ source.data-weave meta.directive.fun.dw storage.type.dw
#   ^ source.data-weave meta.directive.fun.dw
#    ^^^^^^ source.data-weave meta.directive.fun.dw entity.name.function.dw
#          ^ source.data-weave meta.directive.fun.dw
#           ^^^^ source.data-weave meta.directive.fun.dw variable.parameter.dw
#               ^ source.data-weave meta.directive.fun.dw keyword.operator.declaration.dw
#                ^ source.data-weave meta.directive.fun.dw
#                 ^^^^ source.data-weave meta.directive.fun.dw support.class.dw
#                     ^^ source.data-weave meta.directive.fun.dw
#                       ^^^^^ source.data-weave meta.directive.fun.dw variable.parameter.dw
#                            ^ source.data-weave meta.directive.fun.dw keyword.operator.declaration.dw
#                             ^ source.data-weave meta.directive.fun.dw
#                              ^^^ source.data-weave meta.directive.fun.dw support.class.dw
#                                 ^ source.data-weave meta.directive.fun.dw
#                                  ^ source.data-weave meta.directive.fun.dw keyword.operator.declaration.dw
#                                   ^ source.data-weave meta.directive.fun.dw
#                                    ^^^^ source.data-weave meta.directive.fun.dw support.class.dw
#                                        ^ source.data-weave meta.directive.fun.dw
#                                         ^ source.data-weave meta.directive.fun.dw keyword.operator.assignment.dw
#                                          ^ source.data-weave
#                                           ^^^^ source.data-weave constant.language.dw